s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java

```diff
@@ -64,14 +64,25 @@
     @PostMapping(value = "/register", consumes = APPLICATION_JSON_VALUE)
     public void register(
             @RequestBody @Valid RegisterRequest registerRequest,
-            HttpServletRequest request
+            HttpServletRequest request,
+            @RequestParam(required = false) String token
     ) throws RecaptchaException {
         val email = registerRequest.getEmail();
 
         validateRecaptcha(request);
 
         try {
             appUserService.register(registerRequest);
+
+            // IMPORTANT!!! not found invitation shouldn't return 404 for correct user registration
+            try {
+                if (token != null) {
+                    invitationService.confirmInvitation(email, token);
+                    val confirmInvitationEvent = new OnConfirmInvitationEvent(token, LocaleContextHolder.getLocale());
+                    eventPublisher.publishEvent(confirmInvitationEvent);
+                }
+            } catch (Exception e) {}
```

Comment:
And why not log, as I suggested in https://github.com/cyfronet-fid/sat4envi/pull/888#discussion_r524011741?